### PR TITLE
update numba patch to unpin setuptools

### PR DESCRIPTION
**Pull request by cdeepali**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/main/CONTRIBUTING.md)?
- [ ] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/main/doc/)?
- [ ] Did you write any tests to validate this change?

## Description

unpin setuptools in numba patch.

## Review process to land 

1. All tests and other checks must succeed.
2. At least one [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) must review and approve.
3. If any  [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) requests changes, they must be addressed.

### CODE REVIEW

**Review by alexlang74 — APPROVED** [review]

**Comment by cdeepali** [comment]
https://github.com/open-ce/open-ce/issues/703

File: envs/feedstock-patches/numba/0001-update-to-numba-0.56.1-and-support-numpy-1.23.patch [hunk 1]
```diff
@@ -1,14 +1,34 @@
...
+@@ -0,0 +1,12 @@
++diff --git a/setup.py b/setup.py
++index 2a6d12277..20948975f 100644
++--- a/setup.py
+++++ b/setup.py
++@@ -375,7 +375,6 @@ build_requires = ['numpy >={},<{}'.format(min_numpy_build_version,
++ install_requires = [
++     'llvmlite >={},<{}'.format(min_llvmlite_version, max_llvmlite_version),
++     'numpy >={},<{}'.format(min_numpy_run_version, max_numpy_run_version),
++-    'setuptools<60',
```
Comment (npanpaliya) [inline envs/feedstock-patches/numba/0001-update-to-numba-0.56.1-and-support-numpy-1.23.patch:26]: Will numba work with setuptools 63.* when they have mentioned this hard requirement in setup.py?

Comment (cdeepali) [reply envs/feedstock-patches/numba/0001-update-to-numba-0.56.1-and-support-numpy-1.23.patch:26]: Builds and import works. I am going to build snapml and try numba tests too.

Comment (cdeepali) [reply envs/feedstock-patches/numba/0001-update-to-numba-0.56.1-and-support-numpy-1.23.patch:26]: As we are already using setuptools 63.* in OpencE 1.7.2 this is already tested. Just that pip check was failing. So this change will avoid the pip check failure.

Comment (npanpaliya) [reply envs/feedstock-patches/numba/0001-update-to-numba-0.56.1-and-support-numpy-1.23.patch:26]: Okay. Great!

**Review by npanpaliya — APPROVED** [review]

**Pull request merged**